src/LibVLCSharp.WPF/VideoView.cs

```diff
@@ -70,31 +65,73 @@ public override void OnApplyTemplate()
-                if (windowsFormsHost != null)
-                {
-                    ForegroundWindow = new ForegroundWindow(windowsFormsHost)
-                    {
-                        OverlayContent = ViewContent
-                    };
-                }
+                return;
+            }
 
-                Hwnd = (Template.FindName(PART_PlayerView, this) as System.Windows.Forms.Panel)?.Handle ?? IntPtr.Zero;
-                if (Hwnd == IntPtr.Zero)
-                {
-                    Trace.WriteLine("HWND is NULL, aborting...");
-                    return;
-                }
+            if (Template.FindName(PART_PlayerHost, this) is not ControlHwndHost controlHost)
+            {
+                Trace.WriteLine($"Couldn't find {PART_PlayerHost} of type {nameof(ControlHwndHost)}");
+                return;
+            }
 
-                if (MediaPlayer == null)
-                {
-                    Trace.Write("No MediaPlayer is set, aborting...");
-                    return;
-                }
+            _controlHost = controlHost;
+            _controlHost.Visibility = Visibility.Hidden;
+
+            ForegroundWindow = new ForegroundWindow(_controlHost)
+            {
+                OverlayContent = ViewContent
+            };
+
+
+            if (_controlHost.Handle == IntPtr.Zero)
+            {
+                Trace.WriteLine("HWND is NULL, aborting...");
+                return;
+            }
+
+            if (MediaPlayer == null)
+            {
+                Trace.Write("No MediaPlayer is set, aborting...");
+                return;
+            }
+            AttachMediaPlayer(MediaPlayer);
+        }
 
-                MediaPlayer.Hwnd = Hwnd;
+        private void AttachMediaPlayer(MediaPlayer mediaPlayer)
+        {
+            if (mediaPlayer != null && _controlHost != null)
+            {
+                mediaPlayer.Hwnd = _controlHost.Handle;
+                mediaPlayer.Playing += MediaPlayer_Playing;
+                mediaPlayer.Stopped += MediaPlayer_Stopped;
+            }
+        }
+
+        private void DetachMediaPlayer(MediaPlayer mediaPlayer)
+        {
+            if (mediaPlayer != null)
+            {
+                mediaPlayer.Hwnd = IntPtr.Zero;
+                mediaPlayer.Playing -= MediaPlayer_Playing;
+                mediaPlayer.Stopped -= MediaPlayer_Stopped;
+            }
+        }
+
+        private void MediaPlayer_Stopped(object? sender, EventArgs e)
+        {
+            if (_controlHost != null)
+            {
+                _controlHost.Dispatcher.BeginInvoke(new Action(() => _controlHost.Visibility = Visibility.Hidden));
+            }
+        }
+
+        private void MediaPlayer_Playing(object? sender, EventArgs e)
+        {
+            if (_controlHost != null)
+            {
+                _controlHost.Dispatcher.BeginInvoke(new Action(() => _controlHost.Visibility = Visibility.Visible));
```

Comment:
I'm not a huge fan of these as this changes the current behavior and is more a matter of personal taste, and that is something that can be done externally.

Reply:
I see your point, but I can see, access to controlHost is kind of hard... but I did implement this code, because I see the issue when I was working with DarkTheme or custom Theme, so I don't want to see the win32's window background as it was the gray.